velox/functions/prestosql/tests/GreatestLeastTest.cpp

```diff
@@ -181,3 +181,119 @@ TEST_F(GreatestLeastTest, clearNulls) {
+ protected:
+  template <typename T>
+  void runDecimalTest(
+      const std::string& query,
+      const std::vector<VectorPtr>& input,
+      const VectorPtr& output) {
+    auto result = evaluate<FlatVector<T>>(query, makeRowVector(input));
+    test::assertEqualVectors(output, result);
+  }
+
+  const std::vector<std::optional<int128_t>> firstColumnLong_ = {
+      buildInt128(10, 300),
+      buildInt128(-10, 300),
+      buildInt128(200, 300),
+      UnscaledLongDecimal::max().unscaledValue() - 1,
+      UnscaledLongDecimal::min().unscaledValue() + 1,
+      std::nullopt};
+  const std::vector<std::optional<int128_t>> secondColumnLong_ = {
+      buildInt128(-10, 300),
+      buildInt128(10, 300),
+      buildInt128(-200, 300),
+      UnscaledLongDecimal::min().unscaledValue() + 1,
+      UnscaledLongDecimal::max().unscaledValue() - 1,
+      buildInt128(1, 1)};
+  const std::vector<std::optional<int64_t>> firstColumnShort_ = {
+      10000,
+      -10000,
+      20000,
+      UnscaledShortDecimal::max().unscaledValue() - 1,
+      UnscaledShortDecimal::min().unscaledValue() + 1,
+      std::nullopt};
+  const std::vector<std::optional<int64_t>> secondColumnShort_ = {
+      -10000,
+      10000,
+      -20000,
+      UnscaledShortDecimal::min().unscaledValue() + 1,
+      UnscaledShortDecimal::max().unscaledValue() - 1,
+      1};
+
+  const TypePtr longType_ = DECIMAL(38, 10);
+  const TypePtr shortType_ = DECIMAL(10, 4);
+};
+
+TEST_F(GreatestLeastDecimalTest, least) {
+  auto firstColumnLong =
+      makeNullableLongDecimalFlatVector(firstColumnLong_, longType_);
+  auto secondColumnLong =
+      makeNullableLongDecimalFlatVector(secondColumnLong_, longType_);
+  auto expectedLong = makeNullableLongDecimalFlatVector(
+      {buildInt128(-10, 300),
+       buildInt128(-10, 300),
+       buildInt128(-200, 300),
+       UnscaledLongDecimal::min().unscaledValue() + 1,
+       UnscaledLongDecimal::min().unscaledValue() + 1,
+       std::nullopt},
+      longType_);
+  runDecimalTest<UnscaledLongDecimal>(
+      "least(c0)", {firstColumnLong}, firstColumnLong);
+  runDecimalTest<UnscaledLongDecimal>(
+      "least(c0, c1)", {firstColumnLong, secondColumnLong}, expectedLong);
+
+  auto firstColumnShort =
+      makeNullableShortDecimalFlatVector(firstColumnShort_, shortType_);
+  auto secondColumnShort =
+      makeNullableShortDecimalFlatVector(secondColumnShort_, shortType_);
+  auto expectedShort = makeNullableShortDecimalFlatVector(
+      {-10000,
+       -10000,
+       -20000,
+       UnscaledShortDecimal::min().unscaledValue() + 1,
+       UnscaledShortDecimal::min().unscaledValue() + 1,
+       std::nullopt},
+      shortType_);
+  runDecimalTest<UnscaledShortDecimal>(
+      "least(c0)", {firstColumnShort}, firstColumnShort);
+  runDecimalTest<UnscaledShortDecimal>(
+      "least(c0, c1)", {firstColumnShort, secondColumnShort}, expectedShort);
+}
+
+TEST_F(GreatestLeastDecimalTest, greatest) {
```

Comment:
Updated the new tests based on decimal type instead of comparison function to reduce some of the redundancy. Updated tests to use existing GreatestLeastTest class since unscaled values vector is no longer redundant.